test/e2e/flowaggregator_test.go

```diff
@@ -1404,6 +1491,26 @@ func matchSrcAndDstAddress(srcIP string, dstIP string, isDstService bool, isIPv6
 	return srcField, dstField
 }
 
+func createToExternalTestPods(t *testing.T, data *TestData) (*PodIPs, string, *PodIPs, string, *PodIPs, string, *PodIPs, func()) {
+	_, serverIPs, serverCleanupFunc := createAndWaitForPod(t, data, func(name string, ns string, nodeName string, hostNetwork bool) error {
+		return data.createServerPod(name, data.testNamespace, "", serverPodPort, false, true)
+	}, "test-server-", "", data.testNamespace, false)
+
+	// Deploy the client Pods on the control-plane node
+	clientAName, clientAIPs, clientACleanupFunc := createAndWaitForPod(t, data, data.createBusyboxPodOnNode, "test-client-", nodeName(0), data.testNamespace, false)
+	clientBName, clientBIPs, clientBCleanupFunc := createAndWaitForPod(t, data, data.createBusyboxPodOnNode, "test-client-", nodeName(0), data.testNamespace, false)
+	clientCName, clientCIPs, clientCCleanupFunc := createAndWaitForPod(t, data, data.createBusyboxPodOnNode, "test-client-", nodeName(0), data.testNamespace, false)
+
+	cleanupFunc := func() {
```

Comment:
you no longer need to worry about passing clean up function around in recent versions of Go. Instead you can register clean up functions directly with `testing.T`:
```go
t.Cleanup(func() {
		serverCleanupFunc()
		clientACleanupFunc()
		clientBCleanupFunc()
		clientCCleanupFunc()
})
```

I think we should use this pattern when we add / update new test code